### ISSUE TITLE
sub-stories

### ISSUE THREAD

**Issue by mohrm** [mohrm/umklapp_site#47]
Sometimes, in a story you want to tell a substory - like, for example, when you want to add some context to a character. Or someone in the story tells another story. 
Wouldn't it be really cool for umklapp to support such sub-stories? I imagine something like the following:
- a teller can choose to start a new sub-story
- following users will then see the part just before the sub-story and the last part of the sub-story. They can then either continue the sub-story, or again start a sub-story, or finish the sub-story.
- if a sub-story is finished, the next teller does not see only the last sentence of it, but the whole sub-story.
- a story can only be finished if all sub-stories are finished

**Comment by nomeata** [comment]
Ahhhh my brain explodes. What? ;-)

**Comment by mohrm** [comment]
Basically, it is like calling functions and seeing the current call stack :-)

Am 4. Juni 2016 16:44:50 MESZ, schrieb Joachim Breitner notifications@github.com:

> Ahhhh my brain explodes. What? ;-)
> 
> ---
> 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/mohrm/umklapp_site/issues/47#issuecomment-223759371

## 

Mit meinem Hosaka 9000 gesendet

**Comment by mohrm** [comment]
There is also a Simpson's episode which illustrates this: It's called "the seemingly never-ending story" (season 17)

Am 4. Juni 2016 17:18:04 MESZ, schrieb Martin Mohr martin.mohr@kit.edu:

> Basically, it is like calling functions and seeing the current call
> stack :-)
> 
> Am 4. Juni 2016 16:44:50 MESZ, schrieb Joachim Breitner
> notifications@github.com:
> 
> > Ahhhh my brain explodes. What? ;-)
> > 
> > ---
> > 
> > You are receiving this because you authored the thread.
> > Reply to this email directly or view it on GitHub:
> > https://github.com/mohrm/umklapp_site/issues/47#issuecomment-223759371

## 

Mit meinem Hosaka 9000 gesendet

**Comment by ratefuchs** [comment]
I really like the idea of substories.

Before trying to implement it, the specification would need to be refined. For me, what happens when I choose to start a new substory is not exactly clear (am I the first writer of the substory or the last writer of the "upper" story?, ...). Same for finishing a substory.

**Comment by mohrm** [comment]
I think when you start a sub-story, you'd have to give an initial part. The next player would then see the part of your predecessor (as part of the "calling" story) and the initial part of the new story.
Until the substory is finished, every teller sees the last part of the calling story and the part of the substory which was written by the guy before him.
When a substory is finished, the next person sees the whole substory as part to continue.

**Comment by nomeata** [comment]
I am very worried about the complexity in terms UI and getting non-stack-based-language-designers to understand this feature.

**Comment by mohrm** [comment]
yes, I know...just wanted to write it down, so that I do not forget to have written it down. Also, theidea sounded cool initially...

**Comment by mohrm** [comment]
maybe just allow one level of nesting?

**Comment by lohner** [comment]
I think this feature would be rarely and only used by some umklapp nerds. If you really want to add it, I would hide it somewhere in the UI where only the nerds find it. Such that there is no impact on the regular user.

**Comment by mohrm** [comment]
> I think this feature would be rarely and only used by some umklapp
> nerds. 

I don't really think so. Sometimes you really want to open a side-story
and return later to the main story. And you really don't want to finish
the whole side-story yourself. I would designate this feature exactly
for these cases. It would help the story tellers to remain focused since
it makes explicit that you are currently in a side-story and will later
return to the main story.
My analogy with function calls and call stacks was merely to make also
nerds like @nomeata understand what I mean ;-)
I honestly think that this feature will also be understood and used by
non-nerd users, at least if we don't nest sub-stories.
